### PR TITLE
Re enable radar and make it work with SensorSpawnerActor

### PR DESCRIPTION
**Pull request by Blyron**
Enable again Radar and adapt it

### CODE REVIEW

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/Radar.cpp [hunk 1]
```diff
@@ -135,6 +141,7 @@ void ARadar::SendLineTraces(float DeltaTime)
     Rays[i].Hitted = false;
   }
 
+  auto LockedPhysObject = FPhysicsObjectExternalInterface::LockRead(GetWorld()->GetPhysicsScene());
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/Radar.cpp:144]: Did you profiled to verify that this optimizes the time execution time?

Adding a lock could increase the execution time even to add parallelization.

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Util/JsonFileManagerLibrary.cpp [hunk 2]
```diff
@@ -243,3 +243,23 @@ bool UJsonFileManagerLibrary::SaveSensorJsonData(const FString& JsonFilePath,
...
+{
+  TArray<TSharedPtr<FJsonValue>> ArrayValue;
+  TSharedPtr<FJsonObject> JsonDataObject = GetSensorJsonData(JsonFilePath, ArrayValue);
+  bool bWriteSuccess = true;
+  for( auto& var : Rays.GetDetections() )
+  {
+    const TSharedPtr<FJsonObject> RadarDataObject = MakeShared<FJsonObject>();
+    RadarDataObject->SetStringField(TEXT("Frame"), FrameNumber);
+    RadarDataObject->SetStringField(TEXT("Velocity"), FString::SanitizeFloat(var.velocity));
+    RadarDataObject->SetStringField( TEXT("Azimuth"), FString::SanitizeFloat(var.azimuth));
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Util/JsonFileManagerLibrary.cpp:257]:
```suggestion
    RadarDataObject->SetStringField(TEXT("Azimuth"), FString::SanitizeFloat(var.azimuth));
```

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Util/SensorSpawnerActor.cpp [hunk 3]
```diff
@@ -270,3 +285,35 @@ void ASensorSpawnerActor::AttachSensorToActor(AActor* SensorActor)
     }
   }
 }
+
+void ASensorSpawnerActor::DrawRadarSensorPoints() const
+{
+  float velocity_range = VelocityRange; // m/s
+  for(const ASensor* CurrentSensor : SpawnedSensorsArray)
+  {
+    if(const ARadar* RadarSensor = Cast<ARadar>(CurrentSensor))
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Util/SensorSpawnerActor.cpp:294]:
```suggestion
    if(const ARadar* RadarSensor = Cast<ARadar>(CurrentSensor); CurrentSensor)
```
Apply the suggested change to check if `CurrentSensor` is null.
See the next documentation to understand why:
https://www.tutorialspoint.com/cplusplus17-if-statement-with-initializer

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Util/SensorSpawnerActor.cpp [hunk 4]
```diff
@@ -270,3 +285,35 @@ void ASensorSpawnerActor::AttachSensorToActor(AActor* SensorActor)
...
+        FTransform SensorTransform(FRotator(CurrentRot.Pitch + Alt,
+                                            CurrentRot.Yaw + Azi,
+                                            CurrentRot.Roll));
+
+        FWVector = SensorTransform.GetRotation().RotateVector(FWVector);
+
+        float Norm_velocity = detect.velocity / velocity_range; 
+        FColor ColorPoint;
+        ColorPoint.R = FMath::Clamp(Norm_velocity, 0.0f, 1.0f) * 255;
+        ColorPoint.G = FMath::Clamp(  1.0f - abs(Norm_velocity), 0.0f, 1.0f) * 255;
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Util/SensorSpawnerActor.cpp:312]:
```suggestion
        ColorPoint.G = FMath::Clamp( 1.0f - abs(Norm_velocity), 0.0f, 1.0f) * 255;
```

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Util/SensorSpawnerActor.cpp [hunk 5]
```diff
@@ -270,3 +285,35 @@ void ASensorSpawnerActor::AttachSensorToActor(AActor* SensorActor)
...
+        ColorPoint.B = FMath::Clamp(abs(- 1.0f - Norm_velocity), 0.0f, 1.0f) * 255;
+        
+        DrawDebugPoint(GetWorld(), RadarSensor->GetActorLocation() + FWVector , PointSize, ColorPoint, false, 0.06);
```
Comment (xavisolesoft) [inline Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Util/SensorSpawnerActor.cpp:315]:
```suggestion
```

Should we show this debug point always?

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Util/SensorSpawnerActor.cpp (hunk 3 above)
Comment (Blyron) [reply Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Util/SensorSpawnerActor.cpp:294]: At the end of the day is the same, with the initializacion inside of the if statment, if the result of the operation is nullptr, AKA Current sensor is not a Radar, the code inside if will be elipsed

File: Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/Radar.cpp (hunk 1 above)
Comment (Blyron) [reply Unreal/CarlaUnreal/Plugins/Carla/Source/Carla/Sensor/Radar.cpp:144]: We are not attempting to make a optimization yet. Just enabling radar as similar as possible to UE4 code, Jose has verified that LineTraces are slower in UE5 than UE4. We will need to look for a solution

**Review by xavisolesoft — APPROVED** [review]

**Pull request merged**